### PR TITLE
Write the Diagnostic Suppressor 

### LINKED ISSUE #2916: Write the Diagnostic Suppressor

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
fixes #2916

### CODE REVIEW

File: src/Roslyn.Suppressions/Roslyn.Suppressions/Container.cs [hunk 1]
```diff
@@ -30,6 +30,8 @@ namespace SonarLint.VisualStudio.Roslyn.Suppressions
     internal interface IContainer : IDisposable
     {
         ILogger Logger { get; }
+
+        ISuppressionChecker SuppressionChecker { get; }
```
Comment (ugras-ergun-sonarsource) [inline src/Roslyn.Suppressions/Roslyn.Suppressions/Container.cs:34]: Placeholder till implementation is finished

File: src/Roslyn.Suppressions/Roslyn.Suppressions/SuppressionChecker.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions
+{
+    internal interface ISuppressionChecker
```
Comment (ugras-ergun-sonarsource) [inline src/Roslyn.Suppressions/Roslyn.Suppressions/SuppressionChecker.cs:26]: Placeholder till implementation is finished

**Comment by ugras-ergun-sonarsource** [comment]
Code Coverage fails due to not being able to mock SuppressionAnalysisContext and some placeholder code.

File: src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/SonarDiagnosticSuppressorTests.cs [hunk 3]
```diff
@@ -44,11 +49,137 @@ public void Suppressor_ReturnsExpectedSuppressions()
...
+
+            var testSubject = CreateTestSubject();
+
+            var suppressions = testSubject.GetSuppressions(reportedDiagnostics, suppressionContext.Object);
+
+            suppressions.Count().Should().Be(0);
+        }
+
+        [TestMethod]
+        public void GetSuppressions_HasSupressedDiagnostic_ShouldReturnSupressions()
```
Comment (duncanp-sonar) [inline src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/SonarDiagnosticSuppressorTests.cs:73]: Minor: repeated typo in this test name and the ones below - "suppressed" has two "p"s.

**Review by duncanp-sonar — APPROVED** [review]
Just the comment about the typos.

**Pull request merged**